Solaris madvise specifies caddr_t* as first parameter whereas Linux uses
void*.

Potentially file a bug and contribute upstream if accepted.

--- mozjs-38.0.0/mfbt/lz4.c	2015-09-17 13:12:16.000000000 -0700
+++ mozjs-38.0.0/mfbt/lz4.c	2017-03-31 23:03:21.240234123 -0700
@@ -104,7 +104,7 @@
 /**************************************
  Compiler Options
 **************************************/
-#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L)   /* C99 */
+#if defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 199901L) && !defined(__cplusplus)   /* C99 */
 /* "restrict" is a known keyword */
 #else
 #  define restrict /* Disable restrict */
--- mozjs-38.0.0/mfbt/Poison.cpp	2017-03-31 23:03:35.872633653 -0700
+++ mozjs-38.0.0/mfbt/Poison.cpp	2017-03-31 23:10:34.043626305 -0700
@@ -129,7 +129,7 @@
 static bool
 ProbeRegion(uintptr_t aRegion, uintptr_t aSize)
 {
-  if (madvise(reinterpret_cast<void*>(aRegion), aSize, MADV_NORMAL)) {
+  if (madvise(reinterpret_cast<char*>(aRegion), aSize, MADV_NORMAL)) {
     return true;
   } else {
     return false;
